apis/changelogs/proto/v1alpha1/changelog.proto

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2024 The Crossplane Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+import "google/protobuf/struct.proto";
+
+// buf:lint:ignore PACKAGE_DIRECTORY_MATCH
+package changelogs.proto.v1alpha1;
+
+option go_package = "github.com/crossplane/crossplane-runtime/apis/changelogs/proto/v1alpha1";
+
+// ChangeLogService is a service that provides the ability to send change log
+// entries.
+service ChangeLogService {
+  // SendChangeLog sends a change log entry to the change log service.
+  rpc SendChangeLog (SendChangeLogRequest) returns (SendChangeLogResponse) {}
+}
+
+// SendChangeLogRequest represents a single change log entry, with detailed
+// information about the resource that was changed.
+message SendChangeLogRequest {
```

Comment:
I'd consider taking everything that constitutes "a change log" and making it a distinct message (e.g. `message ChangeLog`) that is included in a `SendChangeLogRequest`.

I suspect right now that means we'll just have:

```protobuf
message SendChangeLogRequest {
  ChangeLog log = 1;
}

message ChangeLog {
  // All the fields that are currently in SendChangeLogRequest
}
```

The reason I suggest this is:

* If in future we want to add fields that pertain to the request to send a change log but that aren't _actually part of the change log_ (e.g. request metadata) there'll be a more obvious distinction.
* It seems like it might give you a generated struct that's more appropriate to use as a "container type" for all the things that constitute a change log in the Reconciler code. (See related suggestion in that code.)

Reply:
I have made this change to capture all the change log entry specific details in its own message type, which is the included in the Request type.  If we need to add new fields specific to the request itself in the future, we now have a clean separation for that. Good idea! 

https://github.com/crossplane/crossplane-runtime/pull/754/commits/afb1c6b1148b69d12412523e3133266c8c810ea2#diff-0da98c4e35025a6b55e0791d1d95743104fe96e9bb3d0466e27deda37120ee24